### DISCUSSION TITLE
Unable to quickly review PRs due to broken keyboard

### DISCUSSION THREAD

**Discussion by MFarabi619** [cuhacking/2025#173] [Show and tell]
**Description**

Keyboards are electro-mechanical devices used to interface with other devices known as personal computers which are used by humans to speed up computation. Therefore, keyboards are classified as human interface devices.

I encountered a possible defective key switch issue on my hand-assembled split keyboard. To kick off the debugging marathon, I did `sudo pacman -Syu` which then broke my laptop keyboard too. Therefore, I as a human(hopefully) am now currently unable to interface with my personal computer entirely.

I resorted to shuffling up the switches on my split keyboard, during which I received a phone call. I made the wise decision of taking the call and ended up making the problem worse by forgetting which switch(es) were defective. I must now therefore replace the entire set and pray that this is not a PCB matrix issue; if so, then RIP me.

The key switch happened to conveniently belong to my `Ctrl` key, which is one of my most beloved keys given that I am an unfortunate Emacs doomer.

Due to my keyboard-driven workflow, this is an urgent gating issue. It not only hinders my ability to review PRs, but it also affects my quality of life as I live out of Emacs. It also humbles me to remain with code and think twice before venturing into circuits.

**To Reproduce**
1. Hotswap old linear switches back in.
2. Activate n-key rollover (NKRO).
3. Activate keylogger.
4. Keep smashing all keys to build up a solid dataset.
5. Do statistical analysis to see which keys don't print most often.

**Expected behavior**
My keyboard must stop not working so I can continue running into more bugs.

**Screenshots**
![20241102_140819.jpg](https://github.com/user-attachments/assets/206c737c-7344-409d-b0ce-b52fd05da0f0)

**Additional context**
This issue was created from [my fax machine](https://youtu.be/xvFZjo5PgG0?si=-xBxlm3sL3Wmlfpv).

**Comment by MFarabi619** [comment]
Fixed.

**Reply by JeremyFriesenGitHub** [reply]
wait so are your keyboard issues resolved?

**Reply by MFarabi619** [reply]
Yessir

**Comment by james-yap** [comment]
I was so eager to view the [fabled fax machine](https://youtu.be/xvFZjo5PgG0?si=-xBxlm3sL3Wmlfpv), and unfortunately I played this at full volume on my speakers in the office today...

To atone for my carelessness I must now face hours of awkward silence sitting at my desk, reflecting on my mistake.
